File: src/main/java/com/battletrident/consts/Managers.java
package com.battletrident.consts;

import com.battletrident.commands.CommandManager;
import com.battletrident.games.ring.RingManager;
import com.battletrident.listeners.EventManager;
import com.battletrident.schedulers.ScheduleManager;

public class Managers {
	private static Managers managers;
	public static CommandManager CommandManager;
	public static ScheduleManager ScheduleManager;
	public static EventManager EventManager;
	public static RingManager RingManager;

	private Managers(
		CommandManager c,
		ScheduleManager s,
		EventManager e,
		RingManager r
	) {
		CommandManager = c;
		ScheduleManager = s;
		EventManager = e;
		RingManager = r;
	}

	public static Managers getManagers() {
		if (managers == null) {
			synchronized (Managers.class) {
				if (managers == null) {
					managers = new Managers(
						new CommandManager(),
						new ScheduleManager(),
						new EventManager(),
						new RingManager()
					);
				}
			}
		}

		return managers;
	}
}

File: src/main/kotlin/com/BattleTrident/Managers.kt
package com.BattleTrident

import com.BattleTrident.commands.CommandManager
import com.BattleTrident.games.player.PlayerManager
import com.BattleTrident.games.ring.RingManager
import com.BattleTrident.games.state.GameManager
import com.BattleTrident.listeners.EventManager
import com.BattleTrident.schedulers.ScheduleManager

object Managers {
	lateinit var commandManager: CommandManager
	lateinit var playerManager: PlayerManager
	lateinit var scheduleManager: ScheduleManager
	lateinit var eventManager: EventManager
	lateinit var ringManager: RingManager
	lateinit var gameManager: GameManager
}